src/nupic/research/temporal_memory.py

```diff
@@ -406,19 +459,19 @@ def leastUsedCell(cells, connections, random):
 
 
   @staticmethod
-  def growSynapses(connections, initialPermanence, nDesiredNewSynapes,
-                   prevWinnerCells, random, segment):
+  def growSynapses(connections, random, segment, nDesiredNewSynapes,
+                   prevWinnerCells, initialPermanence):
```

Comment:
Fair enough.

By the way, I don't think of these as "public methods". They're functions that are used internally. Technically these functions are static methods, but that was more of an aesthetic decision. They could be cut-and-pasted outside the class and just be used as functions. (Maybe we'll want to do that in the future instead of typing long function names like `ExtendedTemporalMemory.activatePredictedColumn`)